### PR TITLE
KTOR-7823 add devcontainer

### PR DESCRIPTION
**Pull request by stokado**
**Subsystem**
Infrastructure / dev environment

**Motivation**
Provide a dev container for external contributors. [KTOR-7823](https://youtrack.jetbrains.com/issue/KTOR-7823) Create a dev container for Ktor

**Solution**
Add a Dockerfile and devcontainer configuration.

### CODE REVIEW

**Comment by stokado** [comment]
I was able to build a JVM target using this configuration.
However, `./gradlew check` failed with strange errors (e.g. IOException: Broken pipe).
I am opening this PR to gather feedback.
image

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  build: {
```
Comment (e5l) [inline .devcontainer/devcontainer.json:2]: Could you please add quotes for the JSON keys?

**Review by e5l — CHANGES_REQUESTED** [review]

**Comment by e5l** [comment]
Hey @stokado, thanks for the PR! Could you check one minor? I failed to open this in dev container using extension

**Comment by e5l** [comment]
There is also a warning reported from Gradle:
```
ERROR: JAVA_HOME is set to an invalid directory: /usr/lib/jvm/java-21-openjdk-amd64

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

**Comment by stokado** [comment]
> Hey @stokado, thanks for the PR! Could you check one minor? I failed to open this in dev container using extension

I checked a dev container in IntelliJ IDEA Ultimate and in GitHub Codespaces
What extension did you try?

**Comment by e5l** [comment]
vscode dev containers

**Comment by stokado** [comment]
Alright, I will check vscode extension later

**Comment by stokado** [comment]
I've updated configuration and replaced developer user with a root user
Unfortunately I was not able to build project in vscode extension with non-root user
I hope it's fine :)

I've also added quotes around keys in JSON and updated configuration of JAVA_HOME env variable
There was also 1 problem, why could not build project inside vscode extension, I had to replace Windows line separator in `gradlew` with Unix

I was able to sync project in both IntelliJ and vscode

**Review by e5l — APPROVED** [review]

**Comment by osipxd** [comment]
I've added android build tools installation in a separate PR: https://github.com/ktorio/ktor/pull/4513

**Comment by e5l** [comment]
@stokado, @osipxd thank you for the nice work!

**Pull request merged**